pkg/gardener/client.go

```diff
@@ -34,3 +36,23 @@ func NewRestConfigFromFile(kubeconfigFilePath string) (*restclient.Config, error
 
 	return restConfig, err
 }
+
+const (
+	kubeconfigSecretKey = "config"
+)
+
+// TODO: Use this function in the Runtime Controller's FSM
```

Comment:
Lets investigate it

Reply:
FSM states for OIDC, and RBAC use a function that is defined in the `fsm` package. In the next steps I will refactor this to use the function from the `github.com/kyma-project/infrastructure-manager/pkg/gardener` package.